tests/integration/aic.rs

```diff
@@ -0,0 +1,242 @@
+use cjms::{
+    controllers::aic::AICResponse,
+    models::aic::{AICModel, AIC},
+};
+use serde_json::json;
+use time::OffsetDateTime;
+use uuid::{Uuid, Version};
+
+use crate::utils::{
+    random_ascii_string, send_get_request, send_post_request, send_put_request, spawn_app, TestApp,
+};
+
+#[tokio::test]
+async fn test_aic_get_is_not_allowed() {
+    let app = spawn_app().await;
+    let test_cases = vec!["/aic", "/aic/123"];
+    for path in test_cases {
+        let r = send_get_request(&app, path).await;
+        assert_eq!(r.status(), 405, "Failed on path: {}", path);
+    }
+}
+
+#[tokio::test]
+async fn aic_create_with_bad_data() {
+    let app = spawn_app().await;
+    let test_cases = [
+        json!({
+            "flow_id": random_ascii_string(),
+            "cj_id": 42,
+        }),
+        json!({
+            "flow_id": 42,
+            "cj_id": random_ascii_string(),
+        }),
+    ];
+    for data in test_cases {
+        let r = send_post_request(&app, "/aic", data).await;
+        assert_eq!(r.status(), 400);
+        let response = r.text().await.expect("Failed to get response text.");
+        assert!(response.contains("Json deserialize error"));
+    }
+}
+
+/* Caller sends flowId and CJEvent value and not an AIC value
+    - create a new AIC id
+    - save creation time, expiration time, AIC id, flow ID, CJ event value
+    - return expiration time, and AIC id
+*/
+#[tokio::test]
+async fn aic_create_success() {
+    /* SETUP */
+    let (app, cj_event_value, flow_id, data) = setup_aic_test().await;
+    let model = AICModel {
+        db_pool: &app.db_connection(),
+    };
+    /* CALL */
+    let r = send_post_request(&app, "/aic", data).await;
+    assert_eq!(r.status(), 201);
```

Comment:
Shouldn't this assertion go under the `TEST` heading?

Reply:
yes! and general clean up of all these comments.

Reply:
fbc3bff